test/common/verify_3scale_UIBBT_alerts.go

```diff
@@ -0,0 +1,192 @@
+
+	// Get the kube client and namespace from the shared test context
+	client := ctx.Client
+	// start test env
+	useCluster := true
+	testEnv = &envtest.Environment{
+		UseExistingCluster:       &useCluster,
+		AttachControlPlaneOutput: true,
+	}
+	var err error
+	cfg, err = testEnv.Start()
+	if err != nil {
+		t.Fatalf("could not get start test environment %s", err)
+	}
+
+	installationType, err := GetInstallType(cfg)
+	fmt.Println(installationType)
+	if err != nil {
+		t.Logf("Failed to identify installation type: %v", err)
+		t.FailNow()
+	}
+
+	var namespaceDeployment string
+	var namespaceConfig string
+
+	if installationType == "managed-api" {
+
+		namespaceDeployment = "redhat-rhoam-3scale-operator"
+		namespaceConfig = "redhat-rhoam-3scale"
+
+	} else if installationType == "multitenant-managed-api" {
+
+		namespaceDeployment = "sandbox-rhoam-3scale-operator"
+		namespaceConfig = "sandbox-rhoam-3scale"
+	}
+
+	// Scale down 3scale-operator to 0 pods
+	err = scaleDeployments(context.TODO(), client, namespaceDeployment, "threescale-operator-controller-manager-v2", 0)
+	if err != nil {
+		t.Logf("Failed to scale down 3scale-operator: %v", err)
+		t.FailNow()
+	}
+
+	time.Sleep(1 * time.Minute)
+
+	// Scale down system-app to 0 pods
+	err = scaleDeploymentsConfig(t, "system-app", namespaceConfig, 0, client)
+	if err != nil {
+		t.Logf("Failed to scale down 3scale-operator: %v", err)
+		t.FailNow()
+	}
+
+	// Check that all ThreeScale**UIBBT alerts are firing
+	isThreeScaleUIBBTAlertFiring, err := IsThreeScaleUIBBTAlertFiring(ctx)
+	if err != nil {
+		t.Logf("Failed to check if ThreeScale**UIBBT alert is firing: %v", err)
+	}
+	if !isThreeScaleUIBBTAlertFiring {
+		t.Logf("ThreeScale**UIBBT alert is not firing")
+	}
+
+	// Scale 3scale operator back up
+	err = scaleDeployments(context.TODO(), client, namespaceDeployment, "threescale-operator-controller-manager-v2", 1)
+	if err != nil {
+		t.Logf("Failed to scale up 3scale-operator: %v", err)
+	}
+
+	// Scale 3scale operator back up
+	err = scaleDeploymentsConfig(t, "system-app", namespaceConfig, 1, client)
+	if err != nil {
+		t.Logf("Failed to scale up 3scale-operator: %v", err)
+	}
+}
+
+func scaleDeployments(ctx context.Context, client k8sclient.Client, namespace string, deploymentName string, replicas int32) error {
+
+	deployment := &k8sappsv1.Deployment{}
+	err := client.Get(context.TODO(), k8sclient.ObjectKey{
+		Name:      deploymentName,
+		Namespace: namespace,
+	}, deployment)
+
+	if err != nil {
+		return fmt.Errorf("failed to get deployment %q: %v ", deployment.Name, err)
+	}
+
+	deployment.Spec.Replicas = &replicas
+
+	err = client.Update(ctx, deployment)
+
+	if err != nil {
+		return fmt.Errorf("failed to update Deployment object %v", err)
+	}
+	fmt.Printf("Scaled down deployment %q in namespace %q\n", deploymentName, namespace)
+
+	return nil
+}
+
+func IsThreeScaleUIBBTAlertFiring(ctx *TestingContext) (bool, error) {
```

Comment:
lowercase 'I' should be enough, this is not meant to be called from the outside, isn't it?